lib/Doctrine/DBAL/Query/QueryBuilder.php

```diff
@@ -426,7 +426,7 @@ public function add(string $sqlPartName, $sqlPart, bool $append = false) : self
                 foreach ($sqlPart as $part) {
                     $this->sqlParts[$sqlPartName][] = $part;
                 }
-            } elseif ($isArray && is_array($sqlPart[key($sqlPart)])) {
+            } elseif ($isArray && (is_array($sqlPart[key($sqlPart)]) || is_object($sqlPart[key($sqlPart)]))) {
```

Comment:
I had to add this to properly handle the `QueryPartJoin` object, that's not an associative array anymore.
Anyway, this whole `add()` method is doomed to be removed.